### PR TITLE
Add initial enclave cask

### PR DESCRIPTION
**Pull request by enclave-alistair**
Initial Release of the Enclave homebrew cask.

See https://enclave.io for additional context.  I work at Enclave, and once this is accepted I'll integrate `brew bump-cask-pr` into our CI/CD flow.

---

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask {{cask_file}}` is error-free.
- [x] `brew style --fix {{cask_file}}` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask {{cask_file}}` worked successfully.
- [x] `brew install --cask {{cask_file}}` worked successfully.
- [x] `brew uninstall --cask {{cask_file}}` worked successfully.

### CODE REVIEW

File: Casks/enclave.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  url "https://release.enclave.io/enclave_osx-installer-x64-stable-#{version}.pkg"
+  name "Enclave"
+  desc "Safely build private networks without configs, firewalls or access control lists"
+  homepage "https://enclave.io/"
+
+  livecheck do
+    url "https://install.enclave.io/latest/osx-homebrew-version.txt"
+    strategy :page_match
+    regex(/^(.*)$/)
+  end
```
Comment (SMillerDev) [inline Casks/enclave.rb:14]: https://docs.enclave.io/setup/windows/ seems to have the latest package version as a download link, why not make a similar page for macOS

Comment (enclave-alistair) [reply Casks/enclave.rb:14]: We absolutely will, just waiting on the homebrew cask to be publicly available (as that is intended as the primary deployment method). I can add the section before-hand if you'd rather?

Comment (SMillerDev) [reply Casks/enclave.rb:14]: If it's not released yet it actually shouldn't be a cask AFAIK.

Comment (alistairjevans) [reply Casks/enclave.rb:14]: This is the first stable release on macOS (this is not a beta or anything, I'm aware those should go elsewhere).

**Comment by enclave-alistair** [comment]
This is the first stable release on macOS (this is not a beta or anything,
I'm aware those should go elsewhere).

File: Casks/enclave.rb (hunk 1 above)
Comment (enclave-alistair) [reply Casks/enclave.rb:14]: This is the first stable release on macOS (this is not a beta or anything, I'm aware those should go elsewhere).

*Apologies for the extra notifications, tried to respond on email and it messed up the reply chain.

**Comment by enclave-alistair** [comment]
@vitorgalvao/@miccal, is there anything additional I can do to help progress this PR? Any information you need from us?

File: Casks/enclave.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  sha256 "a5af5020f28328672b89c8f782fda28909a1793e97d340816673c0d1326308ab"
+
+  url "https://release.enclave.io/enclave_osx-installer-x64-stable-#{version}.pkg"
+  name "Enclave"
+  desc "Safely build private networks without configs, firewalls or access control lists"
+  homepage "https://enclave.io/"
+
+  livecheck do
+    url "https://install.enclave.io/latest/osx-homebrew-version.txt"
+    strategy :page_match
```
Comment (vitorgalvao) [inline Casks/enclave.rb:12]: Not needed.

File: Casks/enclave.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    regex(/^(.*)$/)
```
Comment (vitorgalvao) [inline Casks/enclave.rb:13]: Do `regex(/(\d+(?:\.\d+)*)/)`, for consistency with other casks.

File: Casks/enclave.rb (hunk 1 above)
Comment (vitorgalvao) [reply Casks/enclave.rb:14]: > I can add the section before-hand if you'd rather?

We’d rather have the page exist before accepting the cask, yes.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]
In the future, please don’t disable the checkbox that allows edits from maintainers. I could’ve already merged the required changes.

File: Casks/enclave.rb (hunk 2 above)
Comment (enclave-alistair) [reply Casks/enclave.rb]: Removed

File: Casks/enclave.rb (hunk 3 above)
Comment (enclave-alistair) [reply Casks/enclave.rb]: Used new regex.

File: Casks/enclave.rb (hunk 1 above)
Comment (enclave-alistair) [reply Casks/enclave.rb]: macOS install page exists, both for deploy docs and livecheck content.

**Comment by enclave-alistair** [comment]
Thanks @vitorgalvao; all requested changes are done. I did not intentionally uncheck "allow edits from maintainers", but I will make sure it is always checked on future PRs. 👍

**Comment by enclave-alistair** [comment]
Looks like the `test enclave (macos-10.15)` CI stage failed due to a Github error, "GitHub Actions has encountered an internal error when running your job."

This appears to have made the `conclusion` stage get stuck somehow? It's been 'started' for 15 minutes.

**Review by vitorgalvao — APPROVED** [review]

**Pull request merged**